test/plots/ibm-trading.ts

```diff
@@ -1,17 +1,12 @@
 import * as Plot from "@observablehq/plot";
 import * as d3 from "d3";
 
-// This example uses both an interval (to define an ordinal x-scale) and a
-// custom transform to parse the dates from their string representation. This is
-// not a recommended pattern: you should instead parse strings to dates when
-// loading the data, say by applying d3.autoType or calling array.map.
 export async function ibmTrading() {
-  const ibm = await d3.csv<any>("data/ibm.csv").then((data) => data.slice(-20));
+  const ibm = await d3.csv<any>("data/ibm.csv", d3.autoType).then((data) => data.slice(-20));
   return Plot.plot({
     marginBottom: 65,
     x: {
       interval: "day",
-      transform: (d) => d3.utcDay.floor(d3.isoParse(d)),
```

Comment:
This technique no longer works here because, again, the scale **transform** is applied _after_ the mark transform, so if we want the mark transform to apply the scale’s **interval**, we need to parse the dates before the mark transform runs. Since this technique was never recommended (as documented), I think it’s acceptable to change the behavior, and I rewrote this example to use the recommended technique instead.

Reply:
Side note: we could make the previous version of ibm-trading work again if the "day" interval was allowed to read strings (as discussed in #1326).

Reply:
Yes, but as we discussed previously, then `interval: d3.utcDay` would not behave the same as `interval: "day"`, and I don’t think we should go down further the slippery slope of implicit coercion when it’s better to teach users to set their types earlier.